aten/src/ATen/TensorIndexing.h

```diff
@@ -237,23 +238,25 @@ static inline Tensor applySelect(
     int64_t index,
     int64_t real_dim,
     const at::Device& /*self_device*/,
-    const c10::optional<IntArrayRef>& self_sizes) {
+    const c10::optional<SymIntArrayRef>& self_sizes) {
   // See NOTE [nested tensor size for indexing]
   if (self_sizes.has_value()) {
     TORCH_CHECK_INDEX(
         !(index == 0 && dim == 0 && self_sizes->size() == 0),
         "invalid index of a 0-dim tensor. ",
         "Use `tensor.item()` in Python or `tensor.item<T>()` in C++ to convert a 0-dim tensor to a number");
 
-    int64_t size = (*self_sizes)[dim];
+    auto size = (*self_sizes)[dim];
+    // TODO: this guard can be removed once we move the dim wrapping higher.
+    int64_t concrete_size = size.guard_int(__FILE__, __LINE__);
```

Comment:
This guard is a pretty big deal, definitely want to remove it

Reply:
I need to add ` -SymInt` and then we can remove it.

Reply:
@wconstab pushed some dim wrapping code, maybe it is relevant